src/components/Table/Table.tsx

```diff
@@ -119,43 +125,144 @@ const Tbody = styled.tbody`
+}
+
+interface NoSelectionType extends CommonTableProps {
+  selectable?: never;
+  onSelect?: never;
+}
+
+type TableProps = SelectionType | NoSelectionType;
+
 const Table = forwardRef<HTMLTableElement, TableProps>(
-  ({ headers, rows, ...props }, ref) => (
-    <StyledTable
-      ref={ref}
-      {...props}
-    >
-      <THead>
-        <tr>
-          {headers.map((headerProps, index) => (
-            <TableHeader
-              key={`table-header-${index}`}
-              {...headerProps}
-            />
-          ))}
-        </tr>
-      </THead>
-      <Tbody>
-        {rows.map(row => (
-          <TableRow>
-            {row.map(({ label, ...cellProps }, index) => (
-              <TableData {...cellProps}>
-                {headers[index] && <MobileHeader>{headers[index].label}</MobileHeader>}
-                <span>{label}</span>
-              </TableData>
+  ({ headers, rows, selectable, onSelect, ...props }, ref) => (
+    <TableOuterContainer>
+      <MobileActions>
+        {selectable && (
+          <Checkbox
+            label="Select All"
+            checked={rows.every(row => row.checked)}
+            onCheckedChange={(checked: boolean): void => onSelect("all", checked)}
+          />
+        )}
+      </MobileActions>
+      <TableWrapper>
+        <StyledTable
+          ref={ref}
+          {...props}
+        >
+          <THead>
+            <tr>
+              {selectable && (
+                <StyledHeader>
+                  <Checkbox
+                    onCheckedChange={(checked: boolean) =>
+                      onSelect && onSelect("all", checked)
+                    }
+                  />
+                </StyledHeader>
+              )}
+              {headers.map((headerProps, index) => (
+                <TableHeader
+                  key={`table-header-${index}`}
+                  {...headerProps}
+                />
+              ))}
+            </tr>
+          </THead>
+          <Tbody>
+            {rows.map(({ cells, checked, ...rowProps }, rowIndex) => (
+              <TableRow
+                key={`table-body-row-${rowIndex}`}
+                $selectable={selectable}
+                {...rowProps}
+              >
+                {selectable && (
+                  <SelectData>
+                    <Checkbox
+                      checked={checked}
+                      onCheckedChange={(checked: boolean) => onSelect(rowIndex, checked)}
+                    />
+                  </SelectData>
+                )}
+                {cells.map(({ label, ...cellProps }, index) => (
+                  <TableData
+                    key={`table-cell-${index}`}
+                    {...cellProps}
+                  >
+                    {headers[index] && (
+                      <MobileHeader>{headers[index].label}</MobileHeader>
+                    )}
+                    <span>{label}</span>
+                  </TableData>
+                ))}
+              </TableRow>
```

Comment:
let's extract this component. In this way it might be easier to reuse if needed for other use cases

Reply:
@serdec Which part should I extract? I'm not really sure about your comment

Reply:
`236`-`260` just let's make this a standalone component hiding the rendering lofic from the Table component